panacea/query_client_test.go

```diff
@@ -2,285 +2,121 @@ package panacea
+	require.NoError(suite.T(), err)
+	require.GreaterOrEqual(suite.T(), lastTrustedHeight2, lastTrustedHeight)
+}
+
+func (suite *queryClientTestSuite) prepare() (*TrustedBlockInfo, *config.Config) {
+	hash, height, err := integration.QueryLatestBlock(suite.PanaceaEndpoint("http", 1317))
+	require.NoError(suite.T(), err)
 
-	trustedBlockinfo := TrustedBlockInfo{
-		TrustedBlockHeight: 99,
+	trustedBlockInfo := &TrustedBlockInfo{
+		TrustedBlockHeight: height,
 		TrustedBlockHash:   hash,
 	}
 
 	conf := &config.Config{
-		BaseConfig: config.BaseConfig{
-			LogLevel:          "",
-			OracleMnemonic:    "",
-			ListenAddr:        "",
-			Subscriber:        "",
-			DataDir:           "data",
-			OraclePrivKeyFile: "oracle_priv_key.sealed",
-			OraclePubKeyFile:  "oracle_pub_key.json",
-			NodePrivKeyFile:   "node_priv_key.sealed",
-		},
 		Panacea: config.PanaceaConfig{
-			GRPCAddr:                "https://grpc.gopanacea.org:443",
-			RPCAddr:                 "https://rpc.gopanacea.org:443",
-			ChainID:                 "panacea-3",
-			DefaultGasLimit:         200000,
-			DefaultFeeAmount:        "1000000umed",
-			LightClientPrimaryAddr:  "https://rpc.gopanacea.org:443",
-			LightClientWitnessAddrs: []string{"https://rpc.gopanacea.org:443"},
+			GRPCAddr:                suite.PanaceaEndpoint("tcp", 9090),
+			RPCAddr:                 suite.PanaceaEndpoint("tcp", 26657),
+			ChainID:                 suite.chainID,
+			LightClientPrimaryAddr:  suite.PanaceaEndpoint("tcp", 26657),
+			LightClientWitnessAddrs: []string{suite.PanaceaEndpoint("tcp", 26657)},
 		},
 	}
 
-	require.NoError(t, err)
-
-	queryClient, err := NewQueryClient(ctx, conf, trustedBlockinfo)
-
-	require.NoError(t, err)
-
-	mediblocLimitedAddress := "panacea1ewugvs354xput6xydl5cd5tvkzcuymkejekwk3"
-
-	var wg sync.WaitGroup
-	wg.Add(5)
-
-	fmt.Println("query 1 send")
-	go func() {
-		_, err := queryClient.GetAccount(mediblocLimitedAddress)
-		require.NoError(t, err)
-		wg.Done()
-		fmt.Println("query 1 receive")
-	}()
-
-	fmt.Println("query 2 send")
-	go func() {
-		_, err := queryClient.GetAccount(mediblocLimitedAddress)
-		require.NoError(t, err)
-		wg.Done()
-		fmt.Println("query 2 receive")
-	}()
-
-	fmt.Println("query 3 send")
-	go func() {
-		_, err := queryClient.GetAccount(mediblocLimitedAddress)
-		require.NoError(t, err)
-		wg.Done()
-		fmt.Println("query 3 receive")
-	}()
-
-	fmt.Println("query 4 send")
-	go func() {
-		_, err := queryClient.GetAccount(mediblocLimitedAddress)
-		require.NoError(t, err)
-		wg.Done()
-		fmt.Println("query 4 receive")
-	}()
-
-	fmt.Println("query 5 send")
-	go func() {
-		_, err := queryClient.GetAccount(mediblocLimitedAddress)
-		require.NoError(t, err)
-		wg.Done()
-		fmt.Println("query 5 receive")
-	}()
-	wg.Wait()
-
+	return trustedBlockInfo, conf
 }
-
-// The test below is commented out because it is a test targeting local panacea.
-
-// Test for GetOracleRegistration function.
-//func TestGetOracleRegistration(t *testing.T) {
```

Comment:
We don't need to test all helper functions defined in the `query_client.go`. It's good enough to just test the core logic which may be `GetStoreData()`. But, instead, I left the `TestGetAccount()` as it was.